### DISCUSSION TITLE
airfow can not import   RemovedInAirflow3Warning ???

### DISCUSSION THREAD

**Discussion by timecommunication** [apache/airflow#27505] [General]
### Apache Airflow version

2.4.2

### What happened

After upgrading my airflow from 2.3.3 to 2.4.2, I found this error on web UI. 

![屏幕截图 2022-11-04 102658](https://user-images.githubusercontent.com/19165258/199872330-f01d5cb1-c508-4c51-bcb9-91b2a39628d4.png)

### What you think should happen instead

how to fix this error ?

### How to reproduce

upgrade command is :

pip3 install "apache-airflow==2.4.2" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.4.2/constraints-no-providers-3.10.txt"

### Operating System

Ubuntu 22.04.1 LTS

### Versions of Apache Airflow Providers

apache-airflow-providers-common-sql==1.0.0
apache-airflow-providers-ftp==3.0.0
apache-airflow-providers-http==3.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-sqlite==3.0.0

### Deployment

Virtualenv installation

### Deployment details

upgrade command is :

pip3 install "apache-airflow==2.4.2" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.4.2/constraints-no-providers-3.10.txt"

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by timecommunication** [comment]
it seems the error occurs in airflow source code.

**Comment by Taragolis** [comment]
@timecommunication did you restart all airflow services after upgrade airflow version in virtual environment?

**Comment by timecommunication** [comment]
> @timecommunication did you restart all airflow services after upgrade airflow version in virtual environment?

Yes, I did.

I have restarted it many times.

**Reply by Taragolis** [reply]
Also might be useful to try to find airflow related process

```shell
ps aux | grep -i airflow
```

And try to stop services and check if some process still exists

**Comment by potiuk** [comment]
I believe your scheduler/DagFileProcessor is still not upgraded for some reason. Either you made some typo/error when restarting/reconfiguring the scheduler, or maybe the venv you use for the scheduler has an old version of airflow (you can even accidentally downgrade airflow if you install another dependency that new airflow does not support) or maybe your scheduler is run as a service that you forgot to restart. You need to investigate.

Converting to a discussion as this is user problem not airflow bug.

**Reply by potiuk** [reply]
Another option is that you have another "airflow" somewhere on your PYTHONPATH and "airflow.exceptions" is imported from another installation of airflow (this might happen if you have an incorrectly set PYTHONPATH).

You shoudl look at what your `python -m site` shows (when you run it in the same environment as your airflow is) and look if there is `airflow` installed somewhere else than what you expect it to be installed.

**Reply by potiuk** [reply]
Yet another option is that you have "airflow", "airflow.exceptions" folders somewhere in the top-level of your folder and Python looks for the exception in a wrong place. This is not a good idea to do it (if that's the case) - rather than that the DAG "folder" structure is recommended to contain your organisation name at the top level https://airflow.apache.org/docs/apache-airflow/stable/modules_management.html - this is really how Python handles multiple similar package names.

**Reply by handalin** [reply]
@potiuk You are right. By default, Airflow is installed on ~/airflow folder and "~/" is in my PYTHONPATH. I solved this by moving my airflow folder to /opt/airflow (In my case, update the airflow.cfg and set HOME=/opt then restart everything). By the way, setting AIRFLOW_HOME=/opt/airflow didn't work, not sure if the new Airflow refers $HOME instead of $AIRFLOW_HOME or what.
